File: leetcode/387.py
"""
Runtime: 92 ms, faster than 86.13% of Python3 online submissions for First Unique Character in a String.
Memory Usage: 14.7 MB, less than 8.12% of Python3 online submissions for First Unique Character in a String.
"""
from collections import Counter


class Solution:
    def firstUniqChar(self, s: str) -> int:
        cnt = Counter(list(s))
        for i in range(len(s)):
            if cnt[s[i]] == 1:
                return i
        return -1

File: leetcode/387.cpp
// Runtime: 31 ms
// Memory Usage: 11.1 MB
class Solution {
 public:
  int firstUniqChar(string s) {
    std::unordered_map<char, int> count;
    for (const auto& c : s) {
      count[c] += 1;
    }
    for (size_t i = 0; i < s.size(); ++i) {
      if (count[s[i]] == 1) return i;
    }
    return -1;
  }
};
